src/Components/GEOSldas_GridComp/GEOSmetforce_GridComp/LDAS_Forcing.F90

```diff
@@ -5691,8 +5692,20 @@ subroutine get_GEOS_corr_prec_filename(fname_full,file_exists, date_time, met_pa
     
     fname_full_tmp1 = trim(fname_full)                 ! remember for error log below
     
+    ! second try: look for daily file in year/month dir 
+    is = index(fname,'z.nc')
```

Comment:
@gmao-qliu : I think this only works if met_tag does not include "GEOS.fp.asm".  When met_tag includes "GEOS.fp.asm", then fname ends in "_[HHMM].V01.[file_ext].   See if block starting in line 5658 above.  In this case, determining "is" from the position of "z.nc" in the "fname" string won't work.  Since we know the date & time, a better strategy would be start with "fname" and replace [YYYYMMDD_HHMM] with [YYYYMMDD] to construct "fname_tmp"?   Or simply construct fname_tmp in addition to fname within the if block above (line ~5658)?   Having said that, do we even have corrected precip files that use the "GEOS.fp.asm.[...].V01.[file_ext]" template?  In any case, this probably needs more thought and probably a correction.

Reply:
that's good point. Yes we use the GEOS.fp.asm*V01.nc4 format in OPS. Replacing fname would work.

Reply:
"simply construct fname_tmp in addition to fname within the if block above (line ~5658)"   I like this option better as the "replace" one also needs to deal with "z.nc"  and "V01.nc" separately.